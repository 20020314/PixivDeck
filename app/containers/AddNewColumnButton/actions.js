// @flow
import type { User } from '../../types/user'
import type { Action } from './actionTypes.js'
import { ADD_USER_ILLUST } from './constants'

export function addUserIllust(user: User): Action {
  return {
    type: ADD_USER_ILLUST,
    user,
  }
}
