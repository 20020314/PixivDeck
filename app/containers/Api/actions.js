// @flow
import type { Response } from 'services/api'

import { API_REQUEST_SUCCESS } from './actionTypes'
import type { ApiRequestSuccess } from './actionTypes'

export function apiRequestSuccess(response: Response): ApiRequestSuccess {
  return {
    type: API_REQUEST_SUCCESS,
    response,
  }
}
