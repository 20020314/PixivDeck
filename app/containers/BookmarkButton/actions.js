// @flow
import type { Restrict } from './types'

import {
  ADD_BOOKMARK_REQUEST,
  ADD_BOOKMARK_SUCCESS,
  ADD_BOOKMARK_FAILURE,
  DELETE_BOOKMARK_REQUEST,
  DELETE_BOOKMARK_SUCCESS,
  DELETE_BOOKMARK_FAILURE,
} from './actionTypes'
import type {
  AddBookmarkRequest,
  AddBookmarkSuccess,
  AddBookmarkFailure,
  DeleteBookmarkRequest,
  DeleteBookmarkSuccess,
  DeleteBookmarkFailure,
} from './actionTypes'

export function addBookmarkRequest(
  id: number,
  restrict: Restrict
): AddBookmarkRequest {
  return {
    type: ADD_BOOKMARK_REQUEST,
    id,
    restrict,
  }
}
export function addBookmarkSuccess(
  id: number,
  restrict: Restrict
): AddBookmarkSuccess {
  return {
    type: ADD_BOOKMARK_SUCCESS,
    id,
    restrict,
  }
}
export function addBookmarkFailure(
  id: number,
  error: string
): AddBookmarkFailure {
  return {
    type: ADD_BOOKMARK_FAILURE,
    id,
    error,
  }
}
export function deleteBookmarkRequest(id: number): DeleteBookmarkRequest {
  return {
    type: DELETE_BOOKMARK_REQUEST,
    id,
  }
}
export function deleteBookmarkSuccess(id: number): DeleteBookmarkSuccess {
  return {
    type: DELETE_BOOKMARK_SUCCESS,
    id,
  }
}
export function deleteBookmarkFailure(
  id: number,
  error: string
): DeleteBookmarkFailure {
  return {
    type: DELETE_BOOKMARK_FAILURE,
    id,
    error,
  }
}
