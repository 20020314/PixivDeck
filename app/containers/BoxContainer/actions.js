// @flow
import type { Action } from './actionTypes.js'
import { OPEN_PIXIV } from './constants'

export function openPixiv(id: number): Action {
  return {
    type: OPEN_PIXIV,
    id,
  }
}
