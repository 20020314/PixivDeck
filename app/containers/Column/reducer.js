// @flow
import update from 'utils/update'

export type BaseColumn = {
  illustIds: Array<number>,
  nextUrl: ?string,
}

type ActionTypes = {
  ADD_COLUMN_SUCCESS: $Subtype<string>,
  SET_NEXT_URL: $Subtype<string>,
}

export function baseReducer<
  T: BaseColumn,
  S: { [*]: T },
  AS: ActionTypes,
  A: { +type: $Subtype<string> }
>(types: AS, state: S, action: A): S {
  switch (action.type) {
    case types.SET_NEXT_URL:
      return update(state, action, { nextUrl: action.nextUrl })
    case types.ADD_COLUMN_SUCCESS:
      return update(state, action, { illustIds: [], nextUrl: null })
    default:
      return state
  }
}
