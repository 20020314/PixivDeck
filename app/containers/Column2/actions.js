// @flow
import type { Action } from './actionTypes.js'
import { ADD_COLUMN } from './constants'
import type { Body } from './reducer'

export function addColumn(id: string, body: Body): Action {
  return {
    type: ADD_COLUMN,
    id,
    body,
  }
}
