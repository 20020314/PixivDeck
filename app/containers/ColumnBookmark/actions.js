// @flow
import type { Action } from './actionTypes.js'
import {
  ADD_BOOKMARK_COLUMN,
  ADD_BOOKMARK_COLUMN_SUCCESS,
  FETCH_BOOKMARK,
  FETCH_BOOKMARK_FAILRE,
  FETCH_NEXT_BOOKMARK,
  FETCH_NEXT_BOOKMARK_FAILRE,
  SET_NEXT_URL,
  FETCH_BOOKMARK_SUCCESS,
  FETCH_NEXT_BOOKMARK_SUCCESS,
} from './constants'
import type { Response } from 'services/api'
import type { ColumnId } from './reducer'

export function addBookmarkColumn(id: ColumnId): Action {
  return {
    type: ADD_BOOKMARK_COLUMN,
    id,
  }
}

export function addBookmarkColumnSuccess(id: ColumnId): Action {
  return {
    type: ADD_BOOKMARK_COLUMN_SUCCESS,
    id,
  }
}

export function fetchBookmark(id: ColumnId): Action {
  return {
    type: FETCH_BOOKMARK,
    id,
  }
}

export function fetchBookmarkFailre(id: ColumnId): Action {
  return {
    type: FETCH_BOOKMARK_FAILRE,
    id,
  }
}

export function fetchNextBookmark(id: ColumnId): Action {
  return {
    type: FETCH_NEXT_BOOKMARK,
    id,
  }
}

export function fetchNextBookmarkFailre(id: ColumnId): Action {
  return {
    type: FETCH_NEXT_BOOKMARK_FAILRE,
    id,
  }
}

export function setNextUrl(id: ColumnId, nextUrl: string): Action {
  return {
    type: SET_NEXT_URL,
    id,
    nextUrl,
  }
}

export function fetchBookmarkSuccess(
  id: ColumnId,
  response: Response,
  ids: Array<string>
): Action {
  return {
    type: FETCH_BOOKMARK_SUCCESS,
    id,
    response,
    ids,
  }
}

export function fetchNextBookmarkSuccess(
  id: ColumnId,
  response: Response,
  ids: Array<string>
): Action {
  return {
    type: FETCH_NEXT_BOOKMARK_SUCCESS,
    id,
    response,
    ids,
  }
}
