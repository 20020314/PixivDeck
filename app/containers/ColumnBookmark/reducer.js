// @flow
import { handleRehydrate } from 'util/handleReydrate'
import type { Action } from './actionTypes'
import * as Actions from './constants'
import { REHYDRATE } from 'redux-persist/constants'

export type Restrict = 'public' | 'private'

export type Endpoint = '/v1/user/bookmarks/illust'

export type ColumnId = Restrict

export type ColumnBookmark = {|
  illustIds: Array<number>,
  nextUrl: ?string,
|}

export type State = $Shape<{ [ColumnId]: ColumnBookmark }>

const initialState: State = {}

export default function(state: State = initialState, action: Action): State {
  switch (action.type) {
    case Actions.ADD_BOOKMARK_COLUMN_SUCCESS:
      return { ...state, [action.id]: { illustIds: [], nextUrl: null } }

    case Actions.SET_NEXT_URL: {
      const id = action.id
      return {
        ...state,
        [id]: { ...state[id], nextUrl: action.nextUrl },
      }
    }

    case Actions.FETCH_BOOKMARK_SUCCESS:
    case Actions.FETCH_NEXT_BOOKMARK_SUCCESS: {
      const id = action.id
      return {
        ...state,
        [id]: { ...state[id], illustIds: action.ids },
      }
    }

    case REHYDRATE:
      return handleRehydrate(state, action, 'ColumnBookmark')
    default:
      return state
  }
}
