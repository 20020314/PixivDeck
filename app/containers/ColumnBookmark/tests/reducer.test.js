// @flow
import reducer from '../reducer'
import * as actions from '../actions'

const initialState = {}

const response = { result: {}, entities: {} }

test('snapshot initialState', () => {
  // $FlowFixMe
  expect(reducer(initialState, { type: 'default action' })).toMatchSnapshot()
})

test('ADD_BOOKMARK_COLUMN_SUCCESS', () => {
  expect(
    reducer(initialState, actions.addBookmarkColumnSuccess('public'))
  ).toMatchSnapshot()
})

test('SET_NEXT_URL', () => {
  expect(
    reducer(initialState, actions.setNextUrl('public', 'https://next.api.url'))
  ).toMatchSnapshot()
  expect(
    reducer(initialState, actions.setNextUrl('private', 'https://next.api.url'))
  ).toMatchSnapshot()
})

test('FETCH_BOOKMARK_SUCCESS', () => {
  expect(
    reducer(
      initialState,
      actions.fetchBookmarkSuccess('public', response, [1, 2, 3])
    )
  ).toMatchSnapshot()
  expect(
    reducer(
      initialState,
      actions.fetchNextBookmarkSuccess('private', response, [1, 2, 3])
    )
  ).toMatchSnapshot()
})
