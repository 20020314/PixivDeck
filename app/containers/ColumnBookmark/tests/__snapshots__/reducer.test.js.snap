// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ADD_BOOKMARK_COLUMN_SUCCESS 1`] = `
Object {
  "public": Object {
    "illustIds": Array [],
    "nextUrl": null,
  },
}
`;

exports[`FETCH_BOOKMARK_SUCCESS 1`] = `
Object {
  "public": Object {
    "illustIds": Array [
      1,
      2,
      3,
    ],
  },
}
`;

exports[`FETCH_BOOKMARK_SUCCESS 2`] = `
Object {
  "private": Object {
    "illustIds": Array [
      1,
      2,
      3,
    ],
  },
}
`;

exports[`SET_NEXT_URL 1`] = `
Object {
  "public": Object {
    "nextUrl": "https://next.api.url",
  },
}
`;

exports[`SET_NEXT_URL 2`] = `
Object {
  "private": Object {
    "nextUrl": "https://next.api.url",
  },
}
`;

exports[`snapshot initialState 1`] = `Object {}`;
